generator/src/test/java/com/linkedin/pegasus/generator/TestPegasusDataTemplateGenerator.java

```diff
@@ -205,4 +206,62 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
             file -> file.getName().replace(".java", ""),
             Function.identity()));
   }
+
+  @DataProvider(name = "test_schema_permutation_deterministic")
+  private Object[][] createPermutedDataTemplateCases()
+  {
+    return new Object[][]
+        {
+            {"IsoDuration.pdsc", "PremiumService.pdsc", "PremiumService.pdsc", "IsoDuration.pdsc"},
+            {"AField.pdl", "ARecord.pdl", "ARecord.pdl", "AField.pdl"},
+            {"BRecord.pdl", "BField.pdl", "BField.pdl", "BRecord.pdl"},
```

Comment:
I suggest you return two arrays of strings as input. 
Eg
```
{{"IsoDuration.pdsc", "PremiumService.pdsc"}, {"PremiumService.pdsc", "IsoDuration.pdsc"}}
```
And add a comment explaining why order is important for this test.

Reply:
Both of the two points are addressed, I added a comment on the DataProvider, Please let me know if the comment is clear